torch/fx/graph.py

```diff
@@ -153,11 +158,59 @@ def erase_node(self, to_erase : Node):
             raise RuntimeError(f'Tried to erase Node {to_erase} but it still had {len(to_erase.users)} '
                                f'users in the graph: {to_erase.users}!')
 
-        node_indices = [i for i, n in enumerate(self._nodes) if n == to_erase]
-        for idx in reversed(node_indices):
-            self._nodes.pop(idx)
+        to_erase._remove_from_list()
+        to_erase._erased = True  # iterators may retain handles to erased nodes
+        self._len -= 1
+
+    # these can be used to permanently set the insert point: g.insert_before(x)
+    # or when used in a with statement, to temporarily change the insert point:
+    # with g.insert_before(n):
+    #     <do some insertions>
```

Comment:
nit: comment still says `insert_before`

Also, i think the docstring is sufficient here